homeassistant/components/pi_hole/sensor.py

```diff
@@ -1,111 +1,38 @@
-
-ATTR_BLOCKED_DOMAINS = "domains_blocked"
-ATTR_PERCENTAGE_TODAY = "percentage_today"
-ATTR_QUERIES_TODAY = "queries_today"
-
-CONF_LOCATION = "location"
-DEFAULT_HOST = "localhost"
-
-DEFAULT_LOCATION = "admin"
-DEFAULT_METHOD = "GET"
-DEFAULT_NAME = "Pi-Hole"
-DEFAULT_SSL = False
-DEFAULT_VERIFY_SSL = True
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
-
-MONITORED_CONDITIONS = {
-    "ads_blocked_today": ["Ads Blocked Today", "ads", "mdi:close-octagon-outline"],
-    "ads_percentage_today": [
-        "Ads Percentage Blocked Today",
-        "%",
-        "mdi:close-octagon-outline",
-    ],
-    "clients_ever_seen": ["Seen Clients", "clients", "mdi:account-outline"],
-    "dns_queries_today": [
-        "DNS Queries Today",
-        "queries",
-        "mdi:comment-question-outline",
-    ],
-    "domains_being_blocked": ["Domains Blocked", "domains", "mdi:block-helper"],
-    "queries_cached": ["DNS Queries Cached", "queries", "mdi:comment-question-outline"],
-    "queries_forwarded": [
-        "DNS Queries Forwarded",
-        "queries",
-        "mdi:comment-question-outline",
-    ],
-    "unique_clients": ["DNS Unique Clients", "clients", "mdi:account-outline"],
-    "unique_domains": ["DNS Unique Domains", "domains", "mdi:domain"],
-}
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
-        vol.Optional(CONF_LOCATION, default=DEFAULT_LOCATION): cv.string,
-        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=["ads_blocked_today"]): vol.All(
-            cv.ensure_list, [vol.In(MONITORED_CONDITIONS)]
-        ),
-    }
-)
-
-
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Pi-hole sensor."""
-    from hole import Hole
-
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    use_tls = config.get(CONF_SSL)
-    location = config.get(CONF_LOCATION)
-    verify_tls = config.get(CONF_VERIFY_SSL)
-
-    session = async_get_clientsession(hass, verify_tls)
-    pi_hole = PiHoleData(Hole(host, hass.loop, session, location=location, tls=use_tls))
-
-    await pi_hole.async_update()
-
-    if pi_hole.api.data is None:
-        raise PlatformNotReady
-
-    sensors = [
-        PiHoleSensor(pi_hole, name, condition)
-        for condition in config[CONF_MONITORED_CONDITIONS]
-    ]
+
+from .const import DOMAIN as PIHOLE_DOMAIN, ATTR_BLOCKED_DOMAINS, MONITORED_CONDITIONS
+
+LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info):
+    """Set up the pi-hole sensor."""
+    pi_hole = hass.data[PIHOLE_DOMAIN]
+    sensors = []
+
+    # Sometimes the async invocation doesnt contain the entire config
```

Comment:
Instead of doing this, only set up the platform if it has been loaded by the integration. So do this:

```python
if discovery_info is None:
    return
```